blob/driver/driver.go

```diff
@@ -304,3 +306,56 @@ type SignedURLOptions struct {
 	// Expiry sets how long the returned URL is valid for. It is guaranteed to be > 0.
 	Expiry time.Duration
 }
+
+// prefixedBucket implements Bucket by prepending prefix to all keys.
+type prefixedBucket struct {
+	base   Bucket
+	prefix string
+}
+
+// NewPrefixedBucket returns a Bucket based on b with all keys modified to have
+// prefix.
+func NewPrefixedBucket(b Bucket, prefix string) Bucket {
+	return &prefixedBucket{base: b, prefix: prefix}
+}
+
+func (b *prefixedBucket) ErrorCode(err error) gcerrors.ErrorCode { return b.base.ErrorCode(err) }
+func (b *prefixedBucket) As(i interface{}) bool                  { return b.base.As(i) }
+func (b *prefixedBucket) ErrorAs(err error, i interface{}) bool  { return b.base.ErrorAs(err, i) }
+func (b *prefixedBucket) Attributes(ctx context.Context, key string) (*Attributes, error) {
+	return b.base.Attributes(ctx, b.prefix+key)
+}
+func (b *prefixedBucket) ListPaged(ctx context.Context, opts *ListOptions) (*ListPage, error) {
+	var myopts ListOptions
+	if opts != nil {
+		myopts = *opts
+	}
+	myopts.Prefix = b.prefix + myopts.Prefix
+	page, err := b.base.ListPaged(ctx, &myopts)
+	if err != nil {
+		return nil, err
+	}
+	for _, p := range page.Objects {
+		p.Key = strings.TrimPrefix(p.Key, b.prefix)
```

Comment:
IIUC, `p.Key` **should** have a `b.prefix` prefix here. We could `panic` or return an error if it doesn't instead of silently continuing.